Mathlib/Combinatorics/SimpleGraph/Finpartition.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Combinatorics.SimpleGraph.Operations
+import Mathlib.Order.Partition.Finpartition
+
+/-!
+# Dividing the edges of a graph by a vertex finpartition
+
+Consider a finpartition `P` of the vertices of a finite graph. `P` induces a coarser graph,
+whose "supervertices" are `P.parts` and whose "superedges" contain all regular edges whose vertices
+lie in the relevant parts.
+
+Then clearly taking the union over superedges of the contained edges gives back `G.edgeFinset`.
```

Comment:
This is very much not clear if your supergraph is simple, which is what I would have expected (you usually get rid of the edges internal to the parts since they are “bad” and few